lib/internal/process/permission.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const {
+  ObjectFreeze,
+  ArrayIsArray,
+} = primordials;
+
+const permission = internalBinding('permission');
+const path = require('path');
+
+const {
+  ERR_INVALID_ARG_TYPE,
+} = require('internal/errors').codes;
+
+let experimentalPermission;
+
+module.exports = ObjectFreeze({
+  __proto__: null,
+  isEnabled() {
+    if (experimentalPermission === undefined) {
+      const { getOptionValue } = require('internal/options');
+      experimentalPermission = getOptionValue('--experimental-permission');
+    }
+    return experimentalPermission;
+  },
+  deny(scope, references) {
+    if (typeof scope !== 'string')
+      throw new ERR_INVALID_ARG_TYPE('scope', 'string', scope);
```

Comment:
Can we use the validators? If not, can we add a comment explaining why?
```suggestion
    validateString(scope, 'scope');
```